ext/rotoscope/rotoscope.c

```diff
@@ -133,30 +133,24 @@ static rs_tracepoint_t extract_full_tracevals(rb_trace_arg_t *trace_arg,
 
 static bool in_fork(Rotoscope *config) { return config->pid != getpid(); }
 
-static bool tracecmp(rs_tracepoint_t *a, rs_tracepoint_t *b) {
-  return (!rb_str_cmp(a->method_name, b->method_name) &&
-          !rb_str_cmp(a->entity, b->entity) &&
-          a->method_level == b->method_level);
-}
-
-static void log_raw_trace(FILE *stream, rs_tracepoint_t trace) {
+static void log_trace_event(FILE *stream, rs_tracepoint_t *trace) {
   fprintf(stream, RS_CSV_FORMAT "\n", RS_CSV_VALUES(trace));
 }
 
-unsigned char output_buffer[500];
-static void log_stack_frame(FILE *stream, rs_stack_t *stack,
-                            rs_strmemo_t **call_memo, rs_tracepoint_t trace,
-                            rb_event_flag_t event) {
-  if (event & EVENT_CALL) {
-    rs_stack_frame_t frame = rs_stack_push(stack, trace);
-    sprintf((char *)output_buffer, RS_FLATTENED_CSV_FORMAT "\n",
-            RS_FLATTENED_CSV_VALUES(frame));
-
-    if (rs_strmemo_uniq(call_memo, output_buffer)) {
-      fputs((char *)output_buffer, stream);
-    }
-  } else if (event & EVENT_RETURN) {
-    if (tracecmp(&trace, &rs_stack_peek(stack)->tp)) rs_stack_pop(stack);
+unsigned char output_buffer[LOG_BUFFER_SIZE];
+static void log_trace_event_with_caller(FILE *stream,
+                                        rs_stack_frame_t *stack_frame,
+                                        rs_strmemo_t **call_memo) {
+  rs_stack_frame_t *caller = stack_frame->caller;
+  while (caller->blacklisted) {
+    caller = caller->caller;
+  }
```

Comment:
From discussions I've had (cc @stephenminded @airhorns), it sounds like we actually want to print the _true_ caller, even if they are blacklisted; we just want to avoid writing calls into the blacklisted areas. 

Should we make this a config option?

Reply:
A very common case will be calls from tests, aka a blacklisted context. So, for instance:
```
test "whatever" do
   order.do_a_thing
end

class Order
  def do_a_thing
    do_some_other_thing
  end
end
```

Right now, for wedge:
  - I don't care what test called `order.do_a_thing` 
  - I care that `order.do_a_thing` called `some_other_thing`

Potentially we can argue that I do care if the test that called `order.do_a_thing` is in a different component. In that case, having a config option to write what blacklisted context called into a non-blacklisted context (`test "whatever"` in the above example) would be a good idea.

Reply:
Sure, we could easily change it to return the `true` caller.  I think that should be done in another PR though.  It would probably require a bunch of test changes if we just change the default behaviour.  I'm not sure if we want maintain a config option if we only care about the new way for the blacklist to work.